kmath-geometry/src/commonMain/kotlin/scientifik/kmath/geometry/RealEuclideanVectorSpace.kt

```diff
@@ -0,0 +1,74 @@
+package scientifik.kmath.geometry
+
+import scientifik.kmath.dimensions.*
+import scientifik.kmath.operations.ExtendedField
+import scientifik.kmath.operations.RealField
+import kotlin.math.sqrt
+
+data class RealVector<D : Dimension>(private val array: DoubleArray) : Vector<Double, D> {
```

Comment:
No reason to use a data class. Should be private because it is possible to construct a vector with wrong dimensions.